performances/src/main.ts

```diff
@@ -0,0 +1,116 @@
+import puppeteer, { Page } from 'puppeteer'
+
+import { formatProfilingResults } from './format'
+import { startProfiling } from './profiling'
+import { trackNetwork } from './trackNetwork'
+import { ProfilingResults, ProfilingOptions } from './types'
+
+main().catch(console.error)
+
+async function main() {
+  const options: ProfilingOptions = {
+    bundleUrl: 'https://www.datadoghq-browser-agent.com/datadog-rum.js',
+    proxyHost: 'datadog-browser-sdk-profiling-proxy',
+  }
+
+  const wikipediaResults = await profileScenario(options, runWikipediaScenario)
+
+  console.log('# Wikipedia:')
+  console.log()
+  console.log(formatProfilingResults(wikipediaResults))
+  console.log()
+
+  const twitterResults = await profileScenario(options, runTwitterScenario)
+
+  console.log('# Twitter:')
+  console.log()
+  console.log(formatProfilingResults(twitterResults))
+}
+
+async function profileScenario(
+  options: ProfilingOptions,
+  runScenario: (page: Page, takeMeasurements: () => Promise<void>) => Promise<void>
+) {
+  const browser = await puppeteer.launch({
+    defaultViewport: { width: 1366, height: 768 },
+    // Twitter detects headless browsing and refuses to load
+    headless: false,
+  })
+  let result: ProfilingResults
+  try {
+    const page = await browser.newPage()
+    await setupSDK(page, options)
+    const { stopProfiling, takeMeasurements } = await startProfiling(options, page)
+    await runScenario(page, takeMeasurements)
+    result = await stopProfiling()
+  } finally {
+    await browser.close()
+  }
+  return result
+}
+
+async function runWikipediaScenario(page: Page, takeMeasurements: () => Promise<void>) {
+  await page.goto('https://en.wikipedia.org/wiki/Event_monitoring')
+  await takeMeasurements()
+  await page.goto('https://en.wikipedia.org/wiki/Datadog')
+  await takeMeasurements()
+  await page.goto('https://en.wikipedia.org/wiki/Ubuntu')
+  await takeMeasurements()
+  await page.goto('about:blank')
```

Comment:
Is this really needed?

Reply:
Yes, to flush the last page RUM events.

Reply:
In that case, should we take additional measurements after it?

Reply:
Measurements are computed for the current page, at one point in time. On `about:blank`, measurements would always be null, because RUM is not setup on this page.